src/MoveFilesIntoDateDirectories.psm1

```diff
@@ -0,0 +1,78 @@
+#Requires -Version 5.0
+
+function Move-FilesIntoDateDirectories
+{
+	[CmdletBinding()]
+	Param
+	(
+		[Parameter(Mandatory = $true, HelpMessage = 'The directory to look for files to move in.')]
+		[ValidateNotNullOrEmpty()]
+		[System.IO.DirectoryInfo] $SourceDirectoryPath,
+
+		[Parameter(Mandatory = $false, HelpMessage = 'How many subdirectories deep the script should search for files to move. Default is no limit.')]
+		[ValidateRange(0, [int]::MaxValue)]
+		[int] $SourceDirectoryDepthToSearch = [int]::MaxValue,
+
+		[Parameter(Mandatory = $true, HelpMessage = 'The directory to create the date-named directories in and move the files to.')]
+		[ValidateNotNullOrEmpty()]
+		[System.IO.DirectoryInfo] $TargetDirectoryPath,
+
+		[Parameter(Mandatory = $false, HelpMessage = 'The scope at which directories should be created. Accepted values include "Hour", "Day", "Month", or "Year". e.g. If you specify "Day" files will be moved from the `SourceDirectoryPath` to `TargetDirectoryPath\yyyy-MM-dd`.')]
+		[ValidateSet('Hour', 'Day', 'Month', 'Year')]
+		[string] $TargetDirectoriesDateScope = 'Day',
+
+		[Parameter(Mandatory = $false, HelpMessage = 'If provided, the script will overwrite existing files instead of reporting an error the the file already exists.')]
+		[switch] $Force
+	)
+
+	Process
+	{
+		[bool] $sourceDirectoryExists = Test-Path -Path $SourceDirectoryPath -PathType Container
```

Comment:
Added this new block to ensure the source directory exists before attempting to process it